frontend/prebuild/src/app/game/game.component.ts

```diff
@@ -97,14 +100,25 @@ export class GameComponent implements OnInit {
       }
     };
   }
+  
+  ngOnDestroy() {
+    sessionStorage.removeItem('roundId');
+  }
 
   // Game actions
   startGame() {
     this.gameService.send({ message: 'GAME_START' });
   }
 
   requeue() {
-    this.gameService.send({ message: 'GAME_REQUEUE' });
+    if (sessionStorage.getItem('isSpectator') === 'true' ||
+      sessionStorage.getItem('partyId') === null) {
+      this.gameService.send({ message: 'GAME_REQUEUE' });
```

Comment:
Without knowing the implementation details of the backend, is there a reason we need to bounce out to the login page before learning whether the next game is free? Could the message the server sends in response to `GAME_REQUEUE` include a special value or separate field if it doesn't have a roundID because the round is full, and only bounce out to the login route in that case? That could save an extra navigation...

Reply:
yea, this is an unsexy quirk of the backend.  The issue is that sending a `GAME_REQUEUE` message to the gameService (which is tied to a particular round) it always hands you back the next round ID -- regardless of if there are any players waiting in the party queue (because game rounds have no knowledge of parties, and don't care if they are associated with a party or not). 
So the reason I went with the extra navigation was to just have 1 code path for joining a round via party.  I agree the extra navigation isn't very desireable and was planning to revisit this at some point.  Can either do that now or in a separate PR.

Reply:
Nah, don't worry about it now. I suspected "it would twist the backend even worse" was the reason but threw that out there. 

Now that you mention, I think the message protocol has been pretty much left unchanged since we started. I'd also be interested in us reviewing it at some point, time permitting.